### PR TITLE
Add "show company / people" view and "Notes" concept

### PR DESCRIPTION
**Pull request by FelixMalfait**
Ongoing PR / Do not merge.

Goals:
- Convert "commentThread" to a more generic concept that will represent timeline itelines (notes, activity logs, etc.)
- Create a show view for people/company
-  Update design for right drawer

Todo:
- Design a new button to create Notes
- Improve People page (display additional fields?)
- Finish dropdown button (less prio)
- Add more fields in propertybox (less prio)
- Edit from show page (less prio)

Screenshot 2023-07-07 at 17 52 28

### CODE REVIEW

File: front/src/pages/companies/CompanyShow.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+          <Timeline
+            entity={{ id: company?.id ?? '', type: CommentableType.Company }}
+          />
+        </StyledRightPanelContainer>
+      </StyledCompanyContainer>
+    </WithTopBarContainer>
+  );
+}
+
+///////////////////////////////////////////////////////////////
```
Comment (charlesBochet) [inline front/src/pages/companies/CompanyShow.tsx:101]: what happened here? :D

Comment (FelixMalfait) [reply front/src/pages/companies/CompanyShow.tsx:101]: Haha yes I usually do all components in a single file to iterate fast then only split after 😌

File: front/src/modules/ui/components/property-box/PropertyBox.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  children: JSX.Element;
+  extraPadding?: boolean;
+}
+
+export function PropertyBox({
+  children,
+  extraPadding = false,
+}: PropertyBoxProps) {
+  return (
+    <StyledCompanyPropertyBox>
```
Comment (charlesBochet) [inline front/src/modules/ui/components/property-box/PropertyBox.tsx]: Removing Company keyword

File: front/src/modules/ui/components/property-box/PropertyBox.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import styled from '@emotion/styled';
+
+const StyledCompanyPropertyBox = styled.div`
```
Comment (charlesBochet) [inline front/src/modules/ui/components/property-box/PropertyBox.tsx]: looks to be unnecessary container

File: front/src/modules/ui/components/property-box/PropertyBoxItem.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  value: string | ReactNode;
+  link?: string;
+}) {
+  return (
+    <StyledPropertyBoxItem>
+      <StyledLabelAndIconContainer>
+        <StyledIconContainer>{icon}</StyledIconContainer>
+        {label}
+      </StyledLabelAndIconContainer>
+      <StyledValueContainer>
```
Comment (charlesBochet) [inline front/src/modules/ui/components/property-box/PropertyBoxItem.tsx]: I think we should let the parent inject whatever he wants and not care about "link" vs "string" vs "ReactNode". Let's only use ReactNode

File: front/src/modules/ui/components/editor/BlockEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import { BlockNoteEditor } from '@blocknote/core';
+import { BlockNoteView } from '@blocknote/react';
+import { css } from '@emotion/react';
+
+interface BlockEditorProps {
+  editor: BlockNoteEditor | null;
+}
+
+export function BlockEditor({ editor }: BlockEditorProps) {
+  const customStyles = (theme: any) => css`
```
Comment (charlesBochet) [inline front/src/modules/ui/components/editor/BlockEditor.tsx]: Let's keep our pattern with styled components

File: front/src/modules/comments/hooks/useOpenCreateCommentThreadDrawer.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+export function useOpenCreateCommentThreadDrawer() {
+  const openRightDrawer = useOpenRightDrawer();
+
+  const [, setCommentableEntityArray] = useRecoilState(
+    commentableEntityArrayState,
+  );
+
+  return function openCreateCommentThreadDrawer(entity: CommentableEntity) {
+    setCommentableEntityArray([entity]);
+    openRightDrawer('create-comment-thread');
```
Comment (charlesBochet) [inline front/src/modules/comments/hooks/useOpenCreateCommentThreadDrawer.ts]: using enum here

File: front/src/modules/comments/hooks/useOpenCommentThreadRightDrawer.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+export function useOpenCommentThreadRightDrawer() {
+  const openRightDrawer = useOpenRightDrawer();
+  const [, setViewableCommentThreadId] = useRecoilState(
+    viewableCommentThreadIdState,
+  );
+
+  return function openCommentThreadRightDrawer(commentThreadId: string) {
+    setViewableCommentThreadId(commentThreadId);
+    openRightDrawer('edit-comment-thread');
```
Comment (charlesBochet) [inline front/src/modules/comments/hooks/useOpenCommentThreadRightDrawer.ts]: using enum here

File: front/src/modules/comments/components/Timeline.tsx [hunk 8]
```diff
@@ -0,0 +1,288 @@
+import React from 'react';
```
Comment (charlesBochet) [inline front/src/modules/comments/components/Timeline.tsx:1]: I won't refactor this component as very likely to evolve in the upcoming days + tech debt very low as easy to refactor

File: front/src/modules/comments/components/CommentThreadRelationPicker.tsx [hunk 9]
```diff
@@ -22,16 +20,19 @@ import { flatMapAndSortEntityForSelectArrayOfArrayByName } from '@/ui/utils/flat
...
 } from '~/generated/graphql';
 
 import { MultipleEntitySelect } from '../../relation-picker/components/MultipleEntitySelect';
 import { useHandleCheckableCommentThreadTargetChange } from '../hooks/useHandleCheckableCommentThreadTargetChange';
+import { CommentableEntity } from '../types/CommentableEntity';
 import { CommentableEntityForSelect } from '../types/CommentableEntityForSelect';
 
 type OwnProps = {
-  commentThread: CommentThreadForDrawer;
+  commentThread?: GetCommentThreadQuery['findManyCommentThreads'][0];
```
Comment (charlesBochet) [inline front/src/modules/comments/components/CommentThreadRelationPicker.tsx:34]: We shouldn't use query types ; the CommentThreadRelationPicker does not have to know about queries. We shoud use something like CommentThread or even better: Pick<CommentThread, 'id' | 'commentThreadTargets '>

File: front/src/modules/comments/components/CommentThreadEditMode.tsx [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+        commentThreadId: commentThreadId,
+        commentThreadTitle: e.target.value ?? '',
+      },
+      refetchQueries: [getOperationName(GET_COMMENT_THREADS_BY_TARGETS) ?? ''],
+    });
+  }
+
+  const commentThread = data?.findManyCommentThreads[0];
+
+  // ---- Comments
```
Comment (charlesBochet) [inline front/src/modules/comments/components/CommentThreadEditMode.tsx:183]: usually this means we should split the component into pieces!

**Review by charlesBochet — COMMENTED** [review]
First batch ignoring the big new components: CommentThreadEditMode.tsx / CommentThreadCreateMode.tsx / Timeline.tsx

Congrats, huge work!

**Review by charlesBochet — APPROVED** [review]

File: server/src/guards/show-one.guard.ts
Comment (magrinj) [inline server/src/guards/show-one.guard.ts:1]: @FelixMalfait @charlesBochet I think this file was missed

Comment (FelixMalfait) [reply server/src/guards/show-one.guard.ts:1]: I just copied from another file ("create one "was the most similar to "show one" so it's the one I looked at) and assumed it wasn't done anywhere
Should we do both then? I've never done it so maybe more efficient if you take on this?
Thanks!

**Pull request merged**